### ISSUE TITLE
[CSharp] If a string contains a datetime then the deserializer will format it.

### ISSUE THREAD

**Issue by deathly809** [Azure/autorest#2915]
We have a spec where the type is string.  If the value returned is a conforms to a valid datetime then the generated C# code will parse it and change the output.  For example,

In fiddler we see this as the string value,
```
    2018-05-06T19:01:20.7257386Z
```
but when we print out the string we see this
```
    05/09/2018 19:01:20
```

@shahabhijeet Has verified this and provided a fix.  https://github.com/Azure/azure-sdk-for-net/pull/4304

**Comment by deathly809** [comment]
@shahabhijeet So AutoRest does not honor the spec?

**Comment by bganapa** [comment]
Core of the issue is discussed here in the newtonsoft library repo..  https://github.com/JamesNK/Newtonsoft.Json/issues/862

IMO Autorest generated SDK should expose datetime string as string. I am surprised no one has complained about this yet. I see that this will be a breaking change if we modify the autorest behavior now. Consider exposing an option rather than asking the user to modify the generated code.

**Comment by deathly809** [comment]
@bganapa If a user wants to have the string as a DateTime object they can add "format": "datetime".  If I don't add any formatting I expect nothing to ever happen to my strings.

**Comment by bganapa** [comment]
Agree. Autorest should not have converted string to datetime when the spec is not explicitly specifying "format":"datetime". I think autorest inadvertently got this behavior from newtonsoft.
